lib/elixir/lib/kernel.ex

```diff
@@ -261,6 +261,8 @@ defmodule Kernel do
   @doc """
   Returns the head of a list. Raises `ArgumentError` if the list is empty.
 
+  It works with improper lists.
```

Comment:
If it were to me, I wouldn't mention anything about improper lists - most functions don't care if a list is proper or not (unless they traverse it), and it's clear when you understand how a list works. Not sure how we could make it clearer for others.